pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -205,6 +205,16 @@ func newRepositoryWithClient(
 
 // Manifests returns r, which implements distribution.ManifestService.
 func (r *repository) Manifests(ctx context.Context, options ...distribution.ManifestServiceOption) (distribution.ManifestService, error) {
+	if r.pullthrough {
+		// Add to the context the BlobGetterService that provide access to remote servers.
+		// It will be used to validate manifest blobs. It only makes sense
+		// if the pullthrough is enabled.
```

Comment:
Mention that it needs to be instantiated here in order to share the cache among different stat calls made on manifest's dependencies.